### ISSUE TITLE
`pulumi stack history` improvements

### ISSUE THREAD

**Issue by ellismg** [pulumi/pulumi#2412]
We could improve the output of `pulumi history`.  Today, it feels much different than our other commands.

Some ideas:

1. By default have `pulumi history` display a table with a single line per update, if it succeeded or failed and link to the service for that update, as well as the update message.
2. Have `pulumi history <update-id>` show detailed information about a specific update, like tags, the number of resources created and deleted, etc.

**Comment by joeduffy** [comment]
Unless I'm mistaken, `pulumi history` doesn't show version numbers. That's a bit unfortunate since, now that `pulumi stack export` can export a specific version's checkpoint (https://github.com/pulumi/pulumi/pull/3906), these two commands should work well in tandem with one another. As it stands, I'm not sure how I'm supposed to get the version number I'm trying to export? /cc @chrsmith

**Comment by chrsmith** [comment]
Yes. We should add it to the `pulumi history` command, and probably print it on `pulumi stack` too, with the other Pulumi Console-specific metadata.

**Comment by clstokes** [comment]
> Unless I'm mistaken, `pulumi history` doesn't show version numbers. That's a bit unfortunate since, now that `pulumi stack export` can export a specific version's checkpoint (#3906), these two commands should work well in tandem with one another. As it stands, I'm not sure how I'm supposed to get the version number I'm trying to export? /cc @chrsmith

I opened https://github.com/pulumi/pulumi/issues/4328 to address the lack of version number in the output prior to stumbling on to this issue.

**Comment by dixler** [comment]
I've opened the following enhancement which is tangential to this issue
- https://github.com/pulumi/pulumi/issues/12697
